railties/lib/rails/generators/database.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Rails
+  module Generators
+    module Database # :nodoc:
+      DATABASES = %w( mysql postgresql sqlite3 oracle frontbase ibm_db sqlserver )
+      JDBC_DATABASES = %w( jdbcmysql jdbcsqlite3 jdbcpostgresql jdbc )
+      DATABASES.concat(JDBC_DATABASES)
+
+      def initialize(*)
+        super
+        convert_database_option_for_jruby
+      end
+
+      def gem_for_database(database = options[:database])
+        # %w( mysql postgresql sqlite3 oracle frontbase ibm_db sqlserver jdbcmysql jdbcsqlite3 jdbcpostgresql )
+        case database
+        when "mysql"          then ["mysql2", [">= 0.4.4"]]
+        when "postgresql"     then ["pg", [">= 0.18", "< 2.0"]]
+        when "oracle"         then ["activerecord-oracle_enhanced-adapter", nil]
+        when "frontbase"      then ["ruby-frontbase", nil]
+        when "sqlserver"      then ["activerecord-sqlserver-adapter", nil]
+        when "jdbcmysql"      then ["activerecord-jdbcmysql-adapter", nil]
+        when "jdbcsqlite3"    then ["activerecord-jdbcsqlite3-adapter", nil]
+        when "jdbcpostgresql" then ["activerecord-jdbcpostgresql-adapter", nil]
+        when "jdbc"           then ["activerecord-jdbc-adapter", nil]
+        else [database, nil]
```

Comment:
Should this raise an error if we get to the else? Otherwise I think this will silently fail.